File: 1302-delete-characters-to-make-fancy-string/delete-characters-to-make-fancy-string.java
class Solution {
    public String makeFancyString(String s) {
        StringBuilder ans = new StringBuilder();
        int n = s.length();
        if(n<3){
            return s;
        }
        for(int i=0; i<n-3; i++){
            if(s.charAt(i) == s.charAt(i+1) && s.charAt(i+1) == s.charAt(i+2)){
                continue;
            }
            ans.append(s.charAt(i));
        }
        if(s.charAt(n-3) == s.charAt(n-2) && s.charAt(n-2) == s.charAt(n-1)){
            ans.append(s.charAt(n-2));
            ans.append(s.charAt(n-1));
        }else{
            ans.append(s.charAt(n-3));
            ans.append(s.charAt(n-2));
            ans.append(s.charAt(n-1));
        }
        return ans.toString();
    }
}

File: 1302-delete-characters-to-make-fancy-string/delete-characters-to-make-fancy-string.py
class Solution:
    def makeFancyString(self, s: str) -> str:
        if len(s) < 3:
            return s

        n = len(s)
        ans = ""
        
        for i in range(0, n-3):
            if s[i] == s[i+1] and s[i+1] == s[i+2]:
                continue
            ans += s[i]

        if s[n-3] == s[n-2] and s[n-2] == s[n-1]:
            ans += s[n-2:]
        else:
            ans += s[n-3:]

        return ans

        